decidim-core/app/controllers/decidim/user_activities_controller.rb

```diff
@@ -23,14 +23,32 @@ def user
     end
 
     def activities
-      @activities ||= paginate(
-        PublicActivities.new(
-          current_organization,
-          user: user,
-          current_user: current_user,
-          resource_name: filter.resource_type
-        ).query.with_resource_type("all")
-      )
+      @activities ||= begin
+        paginate(
+          activity_class.new(
+            current_organization,
+            user: user,
+            current_user: current_user,
+            resource_name: filter.resource_type
+          ).query.public_send(activity_method, *activity_arguments)
+        )
+      end
+    end
+
+    def activity_method
+      own_activities? ? :with_private_resources : :with_resource_type
```

Comment:
Can we change the non-own activity method to `with_all_resources` and remove the `activity_arguments` which causes this functionality right now?

I found this a bit hard to follow, so I think it would be clearer like that. I think the current approach is just some copy-paste code from some refactor (I know I've at least touched this in the ransack update).

I would rather also make this explicit what is happening here to make it a bit easier to read. Something like:

```ruby
def activities_query
  if own_activities?
    OwnActivities.new(current_organization, **activity_options).with_private_resources
  else
    PublicActivities.new(current_organization, **activity_options).with_all_resources
  end
end

def activity_options
  {
    user: user,
    current_user: current_user,
    resource_name: filter.resource_type
  }
end
```

What do you think?

Reply:
Yeah I tried to make more readable now